File: LongestRepeatingCharacterReplacement.cs
public class LongestRepeatingCharacterReplacement {
    // O(n) time and O(1) space
    public int CharacterReplacement(string s, int k) {
        int left = 0, maxCount = 0, maxLength = 0;
        int[] freq = new int[26];
        for(int right = 0; right < s.Length; right++)
        {
            freq[s[right]-'A']++;
            maxCount = Math.Max(maxCount, freq[s[right]-'A']);

            while((right-left+1)-maxCount > k)
            {
                freq[s[left] - 'A']--;
                left++;
            }
            maxLength = Math.Max(maxLength, right - left+1);
        }
        return maxLength;
    }
}

File: LongestRepeatingCharacterReplacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        #O(n2) time and O(1) space
        res = 0 #O(1) time and O(1) space
        for i in range(len(s)):    #O(n) time and O(1) space
            count,maxf = {},0 #O(1) time and O(1) space
            for j in range(i,len(s)): #O(n) time and O(1) space
                count[s[j]] = 1+count.get(s[j],0)  #O(1) time and O(1) space
                maxf = max(maxf,count[s[j]]) #O(1) time and O(1) space
                if(j-i+1) - maxf <=k: #O(1) time and O(1) space
                    res = max(res,j-i+1) #O(1) time and O(1) space
        return res #O(1) time and O(1) space